### PR TITLE
Add simple command history analyzer

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
image

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

**Review by pokey — APPROVED** [review]
Ok I did some cleanup. Merge if you're happy!

File: packages/cursorless-engine/src/CommandHistoryAnalyzer.ts [hunk 1]
```diff
@@ -5,26 +5,31 @@ import {
...
 import { getPartialTargetDescriptors } from "./util/getPartialTargetDescriptors";
 import { getPartialPrimitiveTargets } from "./util/getPrimitiveTargets";
 import { getScopeType } from "./util/getScopeType";
+import { generateCommandHistoryEntries } from "./generateCommandHistoryEntries";
+import groupby from "./groupby";
+import { mapAsync } from "./mapAsync";
+import { asyncIteratorToList } from "./asyncIteratorToList";
 
-const glob = promisify(globRaw);
+export const glob = promisify(globRaw);
```
Comment (AndreasArvidsson) [inline packages/cursorless-engine/src/CommandHistoryAnalyzer.ts:18]: This feels like a weird export to have in this file. Should we just move this to a common library instead?

Comment (pokey) [reply packages/cursorless-engine/src/CommandHistoryAnalyzer.ts:18]: Nice catch. Happened during auto refactor

**Comment by pokey** [comment]
Ok I did the following:

- Added a totals section before the month-by-month. Tbh I'm not sure the month-by-month is helpful
- Added percentages to each line
- Added a voice command and quick docs blurb
- Added code to handle the case where they have no history

If you're happy feel free to merge

**Review by pokey — APPROVED** [review]

**Pull request merged**